content/azure-pipelines-blueprint-qa-integration.md

```diff
@@ -0,0 +1,114 @@
+```
+
+It is the responsibility of the software delivery team to define **consistent**, **comprehensive**, and **quality** continuous integration tests to empower us to automate the unite and regression testing while building the artifacts we want to deploy. You can extend the [building-code](https://github.com/WorkSafeBC-Common-Engineering/AzureDevOps.Automation.Pipeline.Templates.v2/blob/master/templates/building-code/building-code.yml) to validate the health of your unit tests by using a tool such as [Stryker](https://stryker-mutator.io/).
+
+
+## Continuous delivery (CD) QA extension points?
+
+![CD Integration](../images/azure-pipelines-blueprint-qa-integration-3.png)
+
+The second extension, which runs in parallel to the security automation scans, can be found in the[qa-scans-cd.yml](https://github.com/WorkSafeBC-Common-Engineering/AzureDevOps.Automation.Pipeline.Templates.v2/blob/master/templates/qa/qa-scans-cd.yml) template.
+
+A few words of caution:
+
+- This extension point re-used and triggered by all blueprints!
+- Any tasks running as part of this extension point must be **stable** and **reliable**!
+- This extension point must honour the ```modeElite``` parameter, as discussed below.
+- Validate your extensions using your feature branch and the [ref:](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/resources?view=azure-devops&tabs=schema#define-a-repositories-resource) parameter before committing any change to your master branch. Once in master, your extensions will affect **all** blueprint-based pipelines.
+
+If we peek into the template, we realize that we have two main sections. One that runs for lower (non-production) environments:
+
+```
+# -----------------------------------------------------------------------------------------------------
+# QA AUTOMATION FOR LOWER (NON-PROD) ENVIRONMENTS STAGE
+# -----------------------------------------------------------------------------------------------------
+- ${{ if not(or(eq(variables['Build.SourceBranch'], 'refs/heads/release'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'))) }}:
+```
+
+And another that runs for the higher (production) environments:
+
+```
+# -----------------------------------------------------------------------------------------------------
+# QA AUTOMATION FOR HIGHER (PROD) ENVIRONMENTS STAGE
+# -----------------------------------------------------------------------------------------------------
+- ${{ if or(eq(variables['Build.SourceBranch'], 'refs/heads/release'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release/')) }}:
+```
+
+In other words, you can selectively run quality-assurance in all pipelines or only in pipelines that are based on the ```release/*``` or ```release*``` source code branches, destined for production.
+
+Within both these sections, you will find **blueprint** specific sections, allowing you to your quality assurance automation scans for selected blueprints only.
```

Comment:
allowing you to your quality assurance automation scans for selected blueprints only.
s/b
allowing you to configure your quality assurance automation scans for the selected blueprints only.